flask_resources/views/base.py

```diff
@@ -16,12 +16,20 @@
 
 from ..content_negotiation import content_negotiation
 from ..context import with_resource_requestctx, with_route
+from ..loaders import request_loader
+from ..responses import response_handler
 
 
 class BaseView(MethodView):
     """Base view."""
 
-    resource_decorators = [content_negotiation, with_route, with_resource_requestctx]
+    resource_decorators = [
+        request_loader,
+        response_handler,
+        content_negotiation,
+        with_route,
+        with_resource_requestctx,
+    ]
```

Comment:
major (doubt): The decorators are set in order, right? Then:

1. Content negotiation sets the `accept_mimetype`.
2. Response handler uses it.

```suggestion
    resource_decorators = [
        content_negotiation,
        response_handler,
        request_loader,
        with_route,
        with_resource_requestctx,
    ]
```

no? I also moved loader below, but can be moved up.

Also, the `with_resource_requestctx` is set at the end but is needed for the `content_negotiation`... Which leads me to: Do we (at least me) understand the order in which decorators are applied?

Reply:
Decorators are tricky ヽ(~_~(・_・ )ゝ. Indeed the decorators are **set** in order, but that (or at least our code https://github.com/inveniosoftware/flask-resources/blob/master/flask_resources/views/base.py#L36-L37) leads them to be executed in the other order. What needs to be kept in mind is that *decorators return functions*. They don't execute those functions. They just return them. Here is an illustration:

```
+----------------------------------+
| DecoratorA, DecoratorB           |
+----------------------------------+

1.

   view = DecoratorA(view)


2.

   view = DecoratorB(view) = DecoratorB(DecoratorA(view))

view()
```

So the execution order is: execute DecoratorB, then DecoratorA, the view.: we are good!

Reply:
Perfect thanks :)